### PR TITLE
Add abillity to generate hidden help

### PR DESCRIPTION
**Pull request by rauhul**
- Updates helpMessage(columns:) and helpMessage(for:columns:) with an
  includeHidden argument defaulted to false to allow for clients to
  programmatically generate hidden help.

### CODE REVIEW

**Comment by rauhul** [comment]
@swift-ci please test

File: Tests/ArgumentParserUnitTests/HelpGenerationTests.swift [hunk 1]
```diff
@@ -592,7 +592,7 @@ extension HelpGenerationTests {
 
   func testHelpWithPrivate() {
     // For now, hidden and private have the same behaviour
```
Comment (natecook1000) [inline Tests/ArgumentParserUnitTests/HelpGenerationTests.swift:594]: This comment isn't accurate anymore - does this test still make sense?

**Comment by rauhul** [comment]
@swift-ci please test

File: Tests/ArgumentParserUnitTests/HelpGenerationTests.swift (hunk 1 above)
Comment (rauhul) [reply Tests/ArgumentParserUnitTests/HelpGenerationTests.swift:594]: The test still adds coverage but I think I'll replace these visibility tests in a future PR.

**Comment by natecook1000** [comment]
@swift-ci Please test

**Pull request merged**